packages/vitest/src/node/cli.ts

```diff
@@ -161,6 +161,12 @@ function normalizeCliOptions(argv: CliOptions): CliOptions {
 
     if ((coverage as CoverageIstanbulOptions).ignoreClassMethods)
       (coverage as CoverageIstanbulOptions).ignoreClassMethods = toArray((coverage as CoverageIstanbulOptions).ignoreClassMethods)
+
+    const thresholdAutoUpdate = (coverage as CoverageIstanbulOptions).thresholdAutoUpdate
+    if ((thresholdAutoUpdate as unknown as string) === 'true')
+      (coverage as CoverageIstanbulOptions).thresholdAutoUpdate = true
+    else if ((thresholdAutoUpdate as unknown as string) === 'false')
+      (coverage as CoverageIstanbulOptions).thresholdAutoUpdate = false
```

Comment:
Would this be easier to read and have less type assertions?

```suggestion
    if ('thresholdAutoUpdate' in coverage) {
      // Convert string values ("true"|"false") to boolean
      if (typeof coverage.thresholdAutoUpdate === 'string') {
        const value = coverage.thresholdAutoUpdate === 'false' ? false : coverage.thresholdAutoUpdate

        coverage.thresholdAutoUpdate = value
      }
    }
```

Reply:
I really like the idea but the TypeScript interface suggests Vitest expects a boolean, in the code you suggested, if `thresholdAutoUpdate` is `"true"` it will pass as a string, this could lead to bugs in the future if for some reason we check for exactly `true` instead of just a truthy value.

I have a suggestion that builds on yours:
```suggestion
    if ('thresholdAutoUpdate' in coverage) {
      // Convert string values ('true'|'false') to boolean
      if (typeof coverage.thresholdAutoUpdate === 'string') {
        const value = coverage.thresholdAutoUpdate === 'false'
        ? false
        : coverage.thresholdAutoUpdate === 'true'
        ? true
        : coverage.thresholdAutoUpdate
        
        coverage.thresholdAutoUpdate = value
      }
    }
```
This way the only reason it would pass as anything other than a boolean is due to a user error like using a number.
I also changed the double quotes to single quotes in the comment just because that seems to be what the codebase uses.

What do you think?

Reply:
Looks good 👍